### PR TITLE
Replace Objective-C implemented delegates with Dart implementations

### PR DESCRIPTION
**Pull request by brianquinlan**
- Fixes [#1298](https://github.com/dart-lang/http/issues/1298)

---

- [x] I’ve reviewed the contributor guide and applied the relevant portions to this PR.

<details>
  <summary>Contribution guidelines:</summary><br>

- See our [contributor guide](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md) for general expectations for PRs.
- Larger or significant changes should be discussed in an issue before creating a PR.
- Contributions to our repos should follow the [Dart style guide](https://dart.dev/guides/language/effective-dart) and use `dart format`.
- Most changes should add an entry to the changelog and may need to [rev the pubspec package version](https://github.com/dart-lang/sdk/wiki/External-Package-Maintenance#making-a-change).
- Changes to packages require [corresponding tests](https://github.com/dart-lang/.github/blob/main/CONTRIBUTING.md#Testing).

Note that many Dart repos have a weekly cadence for reviewing PRs - please allow for some latency before initial review feedback.
</details>

### CODE REVIEW

File: pkgs/cupertino_http/lib/src/cupertino_api.dart [hunk 1]
```diff
@@ -1111,279 +793,195 @@ class MutableURLRequest extends URLRequest {
...
+              URLSession._(nsSession,
+                  isBackground: isBackground, hasDelegate: true),
+              URLSessionDownloadTask._(nsTask),
+              nsurlToUri(nsUrl));
+        } finally {
+          nsCondition.unlock();
+        }
+      });
+
+      final downloadDelegate = objc.getProtocol('NSURLSessionDownloadDelegate');
```
Comment (liamappelbe) [inline pkgs/cupertino_http/lib/src/cupertino_api.dart:934]: I've filed https://github.com/dart-lang/native/issues/1675 so you won't have to do this in future.

**Review by liamappelbe — APPROVED** [review]
So much deleted code! LGTM

File: pkgs/cupertino_http/pubspec.yaml [hunk 2]
```diff
@@ -4,10 +4,11 @@ description: >-
   A macOS/iOS Flutter plugin that provides access to the Foundation URL
   Loading System.
 repository: https://github.com/dart-lang/http/tree/master/pkgs/cupertino_http
+publish_to: no
```
Comment (mosuem) [inline pkgs/cupertino_http/pubspec.yaml:7]:
```suggestion
publish_to: none
```

Comment (brianquinlan) [reply pkgs/cupertino_http/pubspec.yaml]: The `publish_to` directive was there by accident.

**Pull request merged**